frappe/patches/v15_0/set_file_type.py

```diff
@@ -0,0 +1,34 @@
+import mimetypes
+
+import frappe
+
+
+def execute():
+	"""Set 'File Type' for all files based on file extension."""
+	files = frappe.db.get_all(
+		"File",
+		fields=["name", "file_name", "is_folder"],
+	)
+
+	frappe.db.auto_commit_on_many_writes = 1
+
+	for file in files:
+		if file.get("is_folder"):
+			continue
+
+		file_extension = get_file_extension(file.get("file_name"))
+		if file_extension:
+			frappe.db.set_value(
+				"File", file.get("name"), "file_type", file_extension, update_modified=False
+			)
```

Comment:
```suggestion
	files = frappe.db.get_all(
		"File",
		fields=["name", "file_name", "file_url"],
		filters={
			"is_folder": 0,
			"file_type": ("is", "not set")
		},
	)

	frappe.db.auto_commit_on_many_writes = 1

	for file in files:
		file_extension = get_file_extension(file.file_name or file.file_url)
		if file_extension:
			frappe.db.set_value(
				"File", file.name, "file_type", file_extension, update_modified=False
			)
```

- Some files have a URL but no file name
- Get only files that are not folders and are missing the file type

Reply:
> Some files have a URL but no file name

Using `file_url` directly to set the file type as the `file_name` is extracted from the last part of the URL.